frmSettings.cs

```diff
@@ -251,5 +257,22 @@ public partial class frmSettings : Form {
                 General.GetRelativeDirectoryPath(path, Settings.ExeDirectory) :
                 General.GetAbsoluteDirectoryPath(path, Settings.ExeDirectory);
         }
+
+        private void ThreadStatusBox_SelectedIndexChanged(object sender, EventArgs e) {
+            if (ThreadStatusBox.Text == "Minutes") {
+                txtThreadStatusBoxThreshold.Enabled = true;
+            }
+            else {
+                txtThreadStatusBoxThreshold.Enabled = false;
+            }
+        }
+
+        private void txtThreadStatusBoxThreshold_Leave(object sender, EventArgs e) {
+            long tmpThreadStatusThreshold;
+            if (!Int64.TryParse(txtThreadStatusBoxThreshold.Text, out tmpThreadStatusThreshold) || tmpThreadStatusThreshold < 0 || tmpThreadStatusThreshold > Int64.MaxValue)
```

Comment:
I think `tmpThreadStatusThreshold` should be an `int` and parsed as such, the same way it is done in the `btnOK_Click` function. No need to check against the type's maximum value as that would have resulted in an incorrect parse.

Reply:
Awesome, thanks - I had looked at the maximum speed text box and used the idea from there; this is cleaner